### ISSUE TITLE
[HOLD for payment 2025-01-17] [$250] GBR in the LHN for an expense that is in pending and have violation

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#52315]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Needs reproduction
**Reproducible in production?:** Needs Reproduction
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation** (hyperlinked to channel name): [ts_external_expensify_expense](https://expensify.slack.com/archives/C06ML6X0W9L/p1731286961448849)

## Action Performed:
Prerequisite: Auto approval limit set for $1000 and Receipt required set for $200
1. Submit a card expense for the amount above $1000 to approver
2. As approver view the expense report

# Expected Result:
If all expenses on a report are pending, don't show the GBR to the approver. 

## Actual Result:
GBR in the LHN for the approver. 

## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

<img src="https://github.com/user-attachments/assets/50656b6f-0653-4c33-bafd-b86f55aa0144" width="40%" height="40%">

<img src="https://github.com/user-attachments/assets/858658a8-0b56-4bb4-be4e-a1ca2c20de37" width="40%" height="40%">

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021858521407289752235</li>
        <li>Upwork Job ID: 1858521407289752235</li>
        <li>Last Price Increase: 2024-11-18</li>
<li>Automatic offers: </li>
<ul>
<li>cretadn22 | Contributor | 105008584</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by trjExpensify** [comment]
@JmillsExpensify this feels like a similar consideration as when all expenses on the report are held and not showing the GBR to the approver in that case. Shall we progress this change independently, or do you want to wrap it up in that one with hold?

**Comment by trjExpensify** [comment]
Ah okay, looking at that issue, I was mistaken thinking we needed to update the GBR logic for hold as well. Sounds like that's in place already (no GBR for the approver when all expenses are held), so we can proceed with this issue for pending transactions independently I think.

**Comment by JmillsExpensify** [comment]
Sounds good!

**Comment by trjExpensify** [comment]
@cretadn22 pending refers to an Expensify Card transaction whereby the transaction is still being processed with the card processor. Here you go: https://github.com/Expensify/App/blob/51a83bf0e2a6fdaab9ba2f06f8fe9828d0c57579/src/libs/TransactionUtils/index.ts#L649-L655

**Comment by trjExpensify** [comment]
@akinwale what do you think of this proposal?

**Comment by akinwale** [comment]
We can move forward with @cretadn22's [proposal](https://github.com/Expensify/App/issues/52315#issuecomment-2486188310) here.

🎀👀🎀 C+ reviewed.

**Comment by JmillsExpensify** [comment]
I'm seeing some weird behavior now with GBR and held expenses, which I just started seeing this week. I wanted to note here, in the hopes that we can fix it in this issue. The issue is what came up and worked at the time, but now it doesn't. Basically this case:

1. Create a workspace, with an admin and a member
2. Have the member create two expenses
3. After creating/submitting the expenses, put one expense on hold

Expected Result: The admin/approver should have GBR for the workspace chat in the LHN, and the approve button should be green. 
Actual Result: The admin/approver has GBR for the workspace chat in the LHN, but the approve button is grey (not green).

**Comment by trjExpensify** [comment]
PR is still in review, Melv.

**Comment by trjExpensify** [comment]
@akinwale can you confirm [this answered your question](https://github.com/Expensify/App/pull/53039#issuecomment-2566118226) and you're able to test the PR?

**Comment by trjExpensify** [comment]
Thanks! Payment summary as follows: 

- $250 to @akinwale for the C+ review (go ahead and request)
- $250 to @cretadn22 for the PR (paid)

Closing!
